### ISSUE TITLE
Two different emphasis are recognized as a single one in a paragraph node

### ISSUE THREAD

**Issue by StanleySathler** [markedjs/marked#1676]
Thank you for the amazing work, guys! :star2: 

**Describe the bug**
Let's suppose the following paragraph:

```
Before explaining about Promises, we should understand what is *async programming*. But, before, let's explain *synchronous programming* in a general context.
```

`Before explaining about Promises, we should understand what is` is consumed correctly. 

The Lexer then calls the method `Tokenizer.em()` which then evaluates the set of `em` rules against the text.

But the last rule (`/^\*([^\s*"<\[][\s\S]*[^\s])\*(?!\*)/`) matches `async programming* . But, before, let's explain *synchronous programming` 

**Expected behavior**
I expected to have matched only `async programming`.

**To Reproduce**
Just try to parse:

```
Before explaining about Promises, we should understand what is *async programming*. But, before, let's explain *synchronous programming* in a general context.
```

**Demo**
1. [Marked Demo](https://marked.js.org/demo/?text=Before%20explaining%20about%20Promises%2C%20we%20should%20understand%20what%20is%20*async%20programming*.%20But%2C%20before%2C%20let%27s%20explain%20*synchronous%20programming*%20in%20a%20general%20context.&options=%7B%0A%20%22baseUrl%22%3A%20null%2C%0A%20%22breaks%22%3A%20false%2C%0A%20%22gfm%22%3A%20true%2C%0A%20%22headerIds%22%3A%20true%2C%0A%20%22headerPrefix%22%3A%20%22%22%2C%0A%20%22highlight%22%3A%20null%2C%0A%20%22langPrefix%22%3A%20%22language-%22%2C%0A%20%22mangle%22%3A%20true%2C%0A%20%22pedantic%22%3A%20false%2C%0A%20%22sanitize%22%3A%20false%2C%0A%20%22sanitizer%22%3A%20null%2C%0A%20%22silent%22%3A%20false%2C%0A%20%22smartLists%22%3A%20false%2C%0A%20%22smartypants%22%3A%20false%2C%0A%20%22tokenizer%22%3A%20null%2C%0A%20%22walkTokens%22%3A%20null%2C%0A%20%22xhtml%22%3A%20false%0A%7D&version=master)
2. [CommonMark Demo](https://spec.commonmark.org/dingus/?text=Before%20explaining%20about%20Promises%2C%20we%20should%20understand%20what%20is%20*async%20programming*.%20But%2C%20before%2C%20let%27s%20explain%20*synchronous%20programming*%20in%20a%20general%20context.)

**Version:** 1.0.0

**Comment by rikedyp** [comment]
Just commenting to say I've bumped into this issue as well. Seems to do with * followed by punctuation as well?

[Marked Demo](https://marked.js.org/demo/?text=*test*%3A%20and%20some%20more%20*stuff*%0A%0AShould%20render%20as%0A%0A%3Cem%3Etest%3C%2Fem%3E%3A%20and%20some%20more%20%3Cem%3Estuff%3C%2Fem%3E&options=%7B%0A%20%22baseUrl%22%3A%20null%2C%0A%20%22breaks%22%3A%20false%2C%0A%20%22gfm%22%3A%20true%2C%0A%20%22headerIds%22%3A%20true%2C%0A%20%22headerPrefix%22%3A%20%22%22%2C%0A%20%22highlight%22%3A%20null%2C%0A%20%22langPrefix%22%3A%20%22language-%22%2C%0A%20%22mangle%22%3A%20true%2C%0A%20%22pedantic%22%3A%20false%2C%0A%20%22sanitize%22%3A%20false%2C%0A%20%22sanitizer%22%3A%20null%2C%0A%20%22silent%22%3A%20false%2C%0A%20%22smartLists%22%3A%20false%2C%0A%20%22smartypants%22%3A%20false%2C%0A%20%22tokenizer%22%3A%20null%2C%0A%20%22walkTokens%22%3A%20null%2C%0A%20%22xhtml%22%3A%20false%0A%7D&version=master)

**Comment by calculuschild** [comment]
Can confirm this is fixed now.

**Comment by UziTech** [comment]
Should be fixed with marked v1.1.1 :tada:
